pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -227,6 +235,84 @@ func (r *ReconcileZookeeperCluster) updateStatefulSet(instance *zookeeperv1beta1
 	return nil
 }
 
+func (r *ReconcileZookeeperCluster) upgradeStatefulSet(instance *zookeeperv1beta1.ZookeeperCluster, foundSts *appsv1.StatefulSet, sts *appsv1.StatefulSet) (err error) {
+
+	//Getting the upgradeCondition from the zk clustercondition
+	_, upgradeCondition := instance.Status.GetClusterCondition(zookeeperv1beta1.ClusterConditionUpgrading)
+
+	//Setting the upgrade condition to true when the zk cluster is upgrading
+	//When the zk cluster is upgrading currentversion is not empty as well as Statefulset CurrentRevision and UpdateRevision are not equal and zk cluster image tag is not equal to CurrentVersion
+	if upgradeCondition != nil && upgradeCondition.Status != corev1.ConditionTrue && instance.Status.IsClusterInReadyState() && foundSts.Status.CurrentRevision != foundSts.Status.UpdateRevision && instance.Spec.Image.Tag != instance.Status.CurrentVersion {
```

Comment:
could we refactor the code in the following way
```
1. upgrade condition = nil
2. upgrade status = false
3. upgrade status = true
```
and add the required steps for each under their respective blocks?

Reply:
I have done this now.